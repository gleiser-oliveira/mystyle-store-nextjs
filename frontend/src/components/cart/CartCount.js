import React from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import { Dot, CountAnimation } from './styles';

const CartCount = props => {
    return (
        <CountAnimation>
            <TransitionGroup>
                <CSSTransition
                    unmountOnExit
                    className="count"
                    classNames="count"
                    key={props.count}
                    timeout={{ enter: 400, exit: 160 }}
                >
                    <Dot>{props.count}</Dot>
                </CSSTransition>
            </TransitionGroup>
        </CountAnimation>
    );
};

CartCount.propTypes = {
    count: PropTypes.number,
};

export default CartCount;
