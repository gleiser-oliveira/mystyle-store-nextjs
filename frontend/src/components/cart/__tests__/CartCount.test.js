import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CartCount from '../CartCount';

describe('Testing <CartCount/> component', () => {
    it('renders', () => {
        shallow(<CartCount count={3} />);
    });

    it('matches the snapshot', () => {
        const wrapper = shallow(<CartCount count={1} />);

        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('updates via props', () => {
        const wrapper = shallow(<CartCount count={1} />);

        expect(toJSON(wrapper)).toMatchSnapshot();

        wrapper.setProps({ count: 3 });

        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
