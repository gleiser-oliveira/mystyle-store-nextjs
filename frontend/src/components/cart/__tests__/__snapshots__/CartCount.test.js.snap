// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CartCount/> component matches the snapshot 1`] = `
<styles__CountAnimation>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="1"
      timeout={
        Object {
          "enter": 400,
          "exit": 160,
        }
      }
      unmountOnExit={true}
    >
      <styles__Dot>
        1
      </styles__Dot>
    </CSSTransition>
  </TransitionGroup>
</styles__CountAnimation>
`;

exports[`Testing <CartCount/> component updates via props 1`] = `
<styles__CountAnimation>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="1"
      timeout={
        Object {
          "enter": 400,
          "exit": 160,
        }
      }
      unmountOnExit={true}
    >
      <styles__Dot>
        1
      </styles__Dot>
    </CSSTransition>
  </TransitionGroup>
</styles__CountAnimation>
`;

exports[`Testing <CartCount/> component updates via props 2`] = `
<styles__CountAnimation>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="3"
      timeout={
        Object {
          "enter": 400,
          "exit": 160,
        }
      }
      unmountOnExit={true}
    >
      <styles__Dot>
        3
      </styles__Dot>
    </CSSTransition>
  </TransitionGroup>
</styles__CountAnimation>
`;
